File: clients/node/src/ui/lobby_screen.js
'use strict';
const { header, hr, BOLD, RESET, DIM, GREEN } = require('./terminal');
const { GAMES } = require('../lobby/session');
const { t } = require('../i18n');

function showLobby(sessions, myName, chatLog) {
  header(t('lobby.title'));
  console.log();
  if (sessions.length) {
    process.stdout.write(BOLD + '  ' + t('lobby.sessions_header') + RESET + '\n');
    sessions.forEach((s, i) => {
      const playerStr = (s.players || []).join(', ');
      console.log('  ' + t('lobby.session_row', { i: i + 1, game: s.game, host: s.host, players: playerStr, port: s.port }));
    });
  } else {
    process.stdout.write(DIM + '  ' + t('lobby.no_sessions') + RESET + '\n');
  }
  console.log();
  hr();
  if (chatLog.length) {
    process.stdout.write(BOLD + '  ' + t('lobby.chat_header') + RESET + '\n');
    for (const e of chatLog.slice(-5)) {
      console.log(`  ${DIM}${e.from}${RESET}: ${e.text}`);
    }
  }
  console.log();
  console.log('  ' + t('lobby.menu'));
  console.log();
}

async function promptHost(myName, { question }) {
  console.log();
  process.stdout.write(BOLD + t('host.games_header') + RESET + '\n');
  const gameList = Object.keys(GAMES).sort();
  gameList.forEach((g, i) => {
    const info = GAMES[g];
    console.log(`  ${DIM}[${i + 1}]${RESET} ${info.name}  (${info.min}~${info.max})`);
  });
  const choice = await question(t('host.pick_game'));
  const idx = parseInt(choice, 10) - 1;
  if (isNaN(idx) || idx < 0 || idx >= gameList.length) return [null, null];
  const game = gameList[idx];
  const maxP = GAMES[game].max || 2;
  const raw = await question(t('host.max_players', { max: maxP }));
  let n = parseInt(raw, 10);
  if (isNaN(n)) n = 2;
  n = Math.min(Math.max(2, n), maxP);
  return [game, n];
}

async function promptJoin(sessions, { question }) {
  const raw = await question(t('host.join_session'));
  const idx = parseInt(raw, 10) - 1;
  if (isNaN(idx) || idx < 0 || idx >= sessions.length) return null;
  return sessions[idx];
}

async function promptChat(myName, { question }) {
  const msg = await question(t('host.chat_prompt', { name: myName }));
  return msg.trim() || null;
}

function renderGame(gameObj, players, chatLog) {
  header(t('game.header'));
  console.log();
  console.log(gameObj.render());
  console.log();
  hr();
  if (chatLog.length) {
    process.stdout.write(BOLD + '  ' + t('game.chat_header') + RESET + '\n');
    for (const e of chatLog.slice(-3)) {
      console.log(`  ${DIM}${e.from}${RESET}: ${e.text}`);
    }
  }
  console.log();
  console.log(`  ${GREEN}${t('game.your_turn')}${RESET}   ${DIM}${t('game.chat_hint')}${RESET}`);
  console.log();
}

module.exports = { showLobby, promptHost, promptJoin, promptChat, renderGame };

File: clients/python/ui/lobby_screen.py
from lobby.session import GAMES
from ui.terminal import header, menu, hr, BOLD, RESET, DIM, CYAN, GREEN, YELLOW


def show_lobby(sessions, my_name, chat_log):
    header('p2p game hub')
    print()
    if sessions:
        print(BOLD + '  Active Sessions' + RESET)
        for i, (sid, s) in enumerate(sessions.items(), 1):
            player_str = ','.join(s.players)
            print(f'  {DIM}[{i}]{RESET} {CYAN}{s.game}{RESET} — host={s.host}  players=[{player_str}]  port={s.port}')
    else:
        print(DIM + '  No sessions found on LAN' + RESET)
    print()
    hr()
    if chat_log:
        print(BOLD + '  Chat' + RESET)
        for entry in chat_log[-5:]:
            print(f'  {DIM}{entry["from"]}{RESET}: {entry["text"]}')
    print()
    print(f'  {DIM}[H]{RESET} Host a game    {DIM}[J]{RESET} Join by number    {DIM}[T]{RESET} Chat    {DIM}[ESC]{RESET} Boss key')
    print()


def prompt_host(my_name):
    print()
    print(BOLD + 'Available games:' + RESET)
    game_list = sorted(GAMES.keys())
    for i, g in enumerate(game_list, 1):
        print(f'  {DIM}[{i}]{RESET} {g}')
    choice = input('Pick game number: ').strip()
    try:
        game = game_list[int(choice) - 1]
    except (ValueError, IndexError):
        return None, None
    max_p = GAMES[game].get('max', 2)
    raw = input(f'Max players (2-{max_p}): ').strip()
    try:
        n = min(max(2, int(raw)), max_p)
    except ValueError:
        n = 2
    return game, n


def prompt_join(sessions):
    keys = list(sessions.keys())
    raw = input('Session number: ').strip()
    try:
        return keys[int(raw) - 1]
    except (ValueError, IndexError):
        return None


def prompt_chat(my_name):
    msg = input(f'{YELLOW}{my_name}{RESET}> ').strip()
    return msg or None


def render_game(game_obj, players, chat_log):
    header('in game')
    print()
    print(game_obj.render())
    print()
    hr()
    if chat_log:
        print(BOLD + '  Chat' + RESET)
        for entry in chat_log[-3:]:
            print(f'  {DIM}{entry["from"]}{RESET}: {entry["text"]}')
    print()
    print(f'  {GREEN}Your move{RESET}   {DIM}[T]{RESET} chat   {DIM}[ESC]{RESET} boss key')
    print()
